src/main/java/com/google/devtools/build/lib/analysis/platform/PlatformInfo.java

```diff
@@ -89,6 +89,8 @@ public class PlatformInfo extends NativeInfo
   private final boolean checkToolchainTypes;
   private final ImmutableList<Label> allowedToolchainTypes;
 
+  @Nullable private final String noToolchainErrorMessage;
```

Comment:
What do you think of calling this "missing toolchain" instead? That makes it obvious that the error is shown as soon as a single toolchain could not be resolved (rather than none out of all relevant toolchain types).

Reply:
Sounds better, thanks.